ompi/info/info.c

```diff
@@ -116,76 +116,85 @@ int ompi_mpiinfo_init(void)
     return OMPI_SUCCESS;
 }
 
-/*
- * Fill in the MPI_INFO_ENV if using MPI3 initialization
+/**
+ * Fill in an info object with ENV info.  Used for setting
+ * MPI_INFO_ENV and by invocation of MPI_Info_create_env.
  */
-int ompi_mpiinfo_init_mpi3(void)
+
+int ompi_mpiinfo_init_env(int argc, char **argv, ompi_info_t *info)
 {
-    char *cptr, **tmp;
+    char *cptr, **tmp = NULL;
 
     /* fill the env info object */
 
     /* command for this app_context */
-    if (NULL != ompi_process_info.command) {
+    if (NULL != argv) {
+        tmp = argv;
+    } else if (NULL != ompi_process_info.command) {
         tmp = opal_argv_split(ompi_process_info.command, ' ');
-        opal_info_set(&ompi_mpi_info_env.info.super, "command", tmp[0]);
+    }
+
+    if (NULL != tmp) {
+        opal_info_set(&info->super, "command", tmp[0]);
 
         /* space-separated list of argv for this command */
         if (1 < opal_argv_count(tmp)) {
             cptr = opal_argv_join(&tmp[1], ' ');
         } else {
             cptr = strdup(tmp[0]);
```

Comment:
@hppritcha `tmp[0]` can also be NULL! 

In fact, if `tmp[0]` is `NULL`, then `opal_argv_count(tmp)` returns zero, and then you endup in this line with `strdup(tmp[0])`.